src/gui/accountstate.cpp

```diff
@@ -230,6 +230,7 @@ bool AccountState::isDesktopNotificationsAllowed() const
 void AccountState::setDesktopNotificationsAllowed(bool isAllowed)
 {
     _isDesktopNotificationsAllowed = isAllowed;
```

Comment:
For extra benefits, please add a check that the value really changed before that line:
```cpp
if (_isDesktopNotificationAllowed == isAllowed) {
    return;
}
```

This way you won't have infinitely recurring signal/slots connection later. All setters in Qt code asks to be implemented that way for peace of mind.